Improve GPS utilization and reporting

JayKickliter opened the pull request:
This PR improves the utilization of our Ublox M8Q GPS by:

1. completely remove NMEA parsing
1. rely solely on UBX messages
1. report additional position and timing information
    1. position accuracy

The second point in the above list is a stepping stone in switching from UART to I2C.

## Compatibility

Even though we're using the M8Q, this u-blox message parsers added in this PR only use the UBX7 versions of `NAV-TIMEGPS` and `NAV-PVT`. On the other hand, this PR does have the downside of adding a hard-requirement on u-blox GPS modules. 

## Remaining Work

- [x] remove temporary hacky logging
- [x] propagate estimated time accuracy
- [x] propagate estimated positional accuracy
- [x] propagate the number of satellites used for fix
- [x] ~rip out all traces of nmea?~ Won't do for the time being
- [x] squash git history. I wasn't able to keep the history as clean as I normally like to, despite liberal use of interactive rebasing

## Questions

Should I rip out NMEA parsing altogether? It is currently doing nothing since I now command the u-blox module to not report NMEA. Having the parsers there also have a knock-on effect in that the packet forwarder currently does pre-parsing to determine whether the bytes it read from the UART are NMEA or UBX protocol in order to dispatch to the correct parser.

Vagabond commented on libloragw/src/loragw_gps.c:
```diff
@@ -640,6 +900,10 @@ int lgw_gps_get(struct timespec *utc, struct timespec *gps_time, struct coord_s
     }
+    if (gps_time_acc != NULL) {
+        gps_time_acc->tv_sec = 0;
+        gps_time_acc->tv_nsec = gps_tAcc_ns;
```
Why?

JayKickliter replied:
Sorry, that's some older cruft that I need to back out. This PR is still WIP. But, that was my original plant to report time accuracy.

Vagabond commented on libloragw/src/loragw_gps.c:
```diff
@@ -415,88 +658,105 @@ enum gps_msg lgw_parse_ubx(const char *serial_buff, size_t buff_size, size_t *ms
-                ck_b = ck_b + ck_a;
-            }
+    if (ubx_sync != 0xB562) {
+        return IGNORED;
```
What happens if the serial output is misaligned, will it eventually resynchronize with the start of a UBX frame?

JayKickliter replied:
I do occasionally see it get out of sync, and it does resynchronize fairly quickly. It's complicated by the fact that this code is processing a mixed stream of binary UBX messages and ASCII NMEA sentences. I currently haven't improved or hurt the situation, and will probably replace the whole thing with a state machine eventually.

Vagabond reviewed (approved):
Looks pretty good and clear.

Does anything get added to the JSON?

JayKickliter commented:
> Does anything get added to the JSON?

It will, but that will live in [`packet_forwarder`](https://github.com/helium/packet_forwarder).

JayKickliter commented:
here's an example of a hiccup. it took a few messages/seconds, but it did recover:

```
NOTE: UBX NAV-PVT
  position is valid
  date is valid
  time is valid
  GPS time = 01:03:21
  msg.valid  0x37
  msg.flags  0x01
  msg.flags2 0x0b
Warning: localization error processing not implemented yet
NOTE: parsed NAV-TIMEGPS, time is valid
  GPS time = 01:03:21
ERROR: UBX message is corrupted, checksum failed
ERROR: UBX message is corrupted, checksum failed
ERROR: UBX message is corrupted, checksum failed
ERROR: UBX message is corrupted, checksum failed
NOTE: UBX NAV-PVT
  position is valid
  date is valid
  time is valid
  GPS time = 01:03:26
  msg.valid  0x37
  msg.flags  0x01
  msg.flags2 0x0b```

fvasquez commented:
> Should I rip out NMEA parsing altogether?

If you do that then you say good bye to the possibility of merging your changes back upstream into Lora-net/lora_gateway. Maybe you've already closed to book on that ever happening. I don't care for NMEA personally but DIY people (eg. Raspbian users) seem to like it.

fvasquez commented on libloragw/library.cfg:
```diff
@@ -9,4 +9,4 @@ DEBUG_SPI= 0
 DEBUG_HAL= 0
 DEBUG_LBT= 0
-DEBUG_GPS= 0
+DEBUG_GPS= 1
```
~Does this spam the log?~ I'm guessing you want to log all the DEBUG_MSG output for the time being.

fvasquez commented on libloragw/src/loragw_gps.c:
```diff
@@ -255,12 +272,36 @@ int lgw_gps_enable(char *tty_path, char *gps_family, speed_t target_brate, int *
+                    0x00, 0x00, /* reserved5 */
+                    0x1A, 0xF9  /* checksum */
+    };
+    _Static_assert(sizeof(ubx_cmd_prt) == 28, "");
```
I didn't know gcc had `_Static_assert` for C.

JayKickliter replied:
It’s in the C11 standard

fvasquez reviewed (approved):
So NMEA over I2C is out of the question? Or just too much of a pain?

JayKickliter committed:
Improve GPS utilization and reporting

- ignore disable NMEA sentances and rely solely on u-blox messages
- propogate positional and timing accuracy estimates
